pint/models/parameter.py

```diff
@@ -144,7 +144,7 @@ def base_value(self):
     def base_value(self,val):
         self._base_value = val
         if not isinstance(val, numbers.Number):
-            return
```

Comment:
Might it not be better to compare vs numbers.Real rather than numbers.Number?  What are the advantages of numbers.Number?